File: implementation/scipamato/scipamato-core-web/src/test/java/ch/difty/scipamato/core/web/user/UserListPageAsViewerTest.java
package ch.difty.scipamato.core.web.user;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.AssertionsForClassTypes.fail;

import org.apache.wicket.authorization.UnauthorizedInstantiationException;
import org.junit.Test;

import ch.difty.scipamato.core.web.WicketTest;

public class UserListPageAsViewerTest extends WicketTest {

    @Override
    protected String getUserName() {
        return "testviewer";
    }

    @Test
    public void viewerCannotAccessUserListPage() {
        try {
            getTester().startPage(new UserListPage(null));
            fail("should have thrown exception");
        } catch (Exception ex) {
            assertThat(ex)
                .isInstanceOf(UnauthorizedInstantiationException.class)
                .hasMessage("Not authorized to instantiate class ch.difty.scipamato.core.web.user.UserListPage");
        }
    }

}

File: core/core-web/src/test/kotlin/ch/difty/scipamato/core/web/user/UserListPageAsViewerTest.kt
package ch.difty.scipamato.core.web.user

import ch.difty.scipamato.core.web.WicketTest
import org.amshove.kluent.invoking
import org.amshove.kluent.shouldThrow
import org.amshove.kluent.withMessage
import org.apache.wicket.authorization.UnauthorizedInstantiationException
import org.junit.jupiter.api.Test

internal open class UserListPageAsViewerTest : WicketTest() {

    override val userName: String
        get() = "testviewer"

    @Test
    fun viewerCannotAccessUserListPage() {
        invoking { tester.startPage(UserListPage(null)) } shouldThrow
            UnauthorizedInstantiationException::class withMessage
            "Not authorized to instantiate class ch.difty.scipamato.core.web.user.UserListPage"
    }
}